cmd/bedMaxWig/bedMaxWig.go

```diff
@@ -42,16 +42,14 @@ func bedMaxWig(infile string, database string, chromsizeFile string, outfile str
 				if currentBed.FieldsInitialized < 7 {
 					currentBed.FieldsInitialized = 7
 				}
+				maxWig := bedRangeMax(chromSlice, records[k].ChromStart, records[k].ChromEnd)
 				if norm == true {
-					maxWig := bedRangeMax(chromSlice, records[k].ChromStart, records[k].ChromEnd)
-					normMaxWig := maxWig / wigTotal
-					currentBed.Annotation = append(currentBed.Annotation, fmt.Sprintf("%f", float64(normMaxWig))) // %f will round to 6th decimal place
-					outlist = append(outlist, currentBed)
-				} else {
-					currentBed.Annotation = append(currentBed.Annotation, fmt.Sprintf("%f", bedRangeMax(chromSlice, records[k].ChromStart, records[k].ChromEnd)))
-					outlist = append(outlist, currentBed)
-
+					maxWig = maxWig / wigTotal
+					currentBed.Annotation = append(currentBed.Annotation, fmt.Sprintf("%e", float64(maxWig))) // %e will have 6 decimals in scientific notation (eg; 1.234456e+78)
+				}else {
+					currentBed.Annotation = append(currentBed.Annotation, fmt.Sprintf("%f", float64(maxWig))) // %f will round to 6th decimal place
```

Comment:
I tried to combine lines 47 through 50 but having the output for non-normalized bedMaxWig seemed difficult to read. Didnt test it, but I think that the exponential way takes up more bytes too?

Like seeing:
chr3    1       2       trf     52      .       1.100000e+01
instead of:
chr3    1       2       trf     52      .       11.000000

Reply:
You could consider merging lines 48 and 50 into a single line that uses %g